pyproject.toml

```diff
@@ -31,3 +31,16 @@ Repository = "https://github.com/data-apis/array-api-strict"
 [tool.setuptools_scm]
 version_file = "array_api_strict/_version.py"
 
+[tool.mypy]
+disallow_incomplete_defs = true
+disallow_untyped_decorators = true
+disallow_untyped_defs = true
+no_implicit_optional = true
+show_error_codes = true
+warn_redundant_casts = true
+warn_unused_ignores = true
+warn_unreachable = true
```

Comment:
I would also recommend `strict_bytes = true`, which disables the "promotion" of `memoryview` and `bytesarray` to `bytes` (similar to how `int` gets promoted to `float` and `float` to `complex`). This will be enabled by default in mypy 2.0.

It might also be a good idea to set `local_partial_types = true`, which will also be the default in mypy 2.0.

Setting `strict = true` is also a good idea (and it isn't as strict as the name suggests it is).

Reply:
I'd rather treat these settings as a good starting point (I copy-pasted them from my template project) and leave tweaking them to a follow-up PR. There are many unfixed issues with them already.

Reply:
I don't expect that enabling `strict_bytes` and `local_partial_types` will lead to different outcomes in this project.